sources/packages/backend/apps/queue-consumers/src/services/cas-supplier/cas-supplier.service.ts

```diff
@@ -72,137 +86,129 @@ export class CASSupplierIntegrationService {
-      if (activeSupplierItemAddress) {
-        summary.info("Updating CAS supplier table.");
-        try {
-          const updateResult = await this.updateCASSupplier(
-            casSupplier.id,
-            supplierInfo,
-            SupplierStatus.Verified,
-          );
-          if (updateResult.affected) {
-            return true;
-          }
-        } catch (error: unknown) {
-          throw new Error(
-            "Unexpected error while updating CAS supplier table.",
-            error,
-          );
-        }
-      } else {
-        summary.info("No active supplier address found on CAS.");
-      }
+  private getCASSupplierProcess(
+    status: CASEvaluationStatus,
+  ): CASEvaluationResultProcessor {
+    switch (status) {
+      case CASEvaluationStatus.PreValidationsFailed:
+        return this.casPreValidationsProcessor;
+      case CASEvaluationStatus.ActiveSupplierFound:
+        return this.casActiveSupplierFoundProcessor;
+      case CASEvaluationStatus.NotFound:
+        return this.casActiveSupplierNotFoundProcessor;
+      default:
+        throw new Error("Invalid CAS evaluation result status.");
     }
-    return false;
   }
 
   /**
-   * Gets the first active supplier address from a list supplier response item.
-   * @param casSupplierResponseItem CAS supplier response item.
-   * @returns CAS supplier response item address.
+   * Decide the current state of the student supplier on SIMS
+   * and return the next process to be executed.
+   * @param casSupplier student CAS supplier to be evaluated.
+   * @param auth authentication token needed for possible
+   * CAS API interactions.
+   * @returns evaluation result to be processed next.
    */
-  private getActiveSupplierItemAddress(
-    casSupplierResponseItem: CASSupplierResponseItem,
-  ): CASSupplierResponseItemAddress {
-    if (casSupplierResponseItem.status !== "ACTIVE") {
-      return undefined;
+  private async evaluateCASSupplier(
+    casSupplier: CASSupplier,
+    auth: CASAuthDetails,
+  ): Promise<CASEvaluationResult> {
+    const preValidationsFailedReasons: PreValidationsFailedReason[] = [];
+    if (!casSupplier.student.user.firstName) {
+      preValidationsFailedReasons.push(
+        PreValidationsFailedReason.GivenNamesNotPresent,
+      );
     }
-    return casSupplierResponseItem.supplieraddress.find(
-      (address) => address.status === "ACTIVE",
+    if (!isAddressFromCanada(casSupplier.student.contactInfo.address)) {
+      preValidationsFailedReasons.push(
+        PreValidationsFailedReason.NonCanadianAddress,
+      );
+    }
+    if (preValidationsFailedReasons.length) {
+      return {
+        status: CASEvaluationStatus.PreValidationsFailed,
+        reasons: preValidationsFailedReasons,
+      };
+    }
+    const supplierResponse = await this.casService.getSupplierInfoFromCAS(
+      auth.access_token,
+      casSupplier.student.sinValidation.sin,
+      casSupplier.student.user.lastName,
     );
```

Comment:
I would recommend, logon and getSupplier call happening here instead upfront. May be a private method which does both together could be better,

Reply:
The logon may not be used exclusively for the `getSupplierInfoFromCAS` and `getSupplierInfoFromCAS` may not even be the first method to be called in the future.
To be clear, is the recommendation to execute the `logon` inside the `evaluateCASSupplier`?

Reply:
To be clear, is the recommendation to execute the logon inside the evaluateCASSupplier? - Yes.

Reply:
The current logon process (not part of this PR) is intended to be executed once and reused where it is needed.
We can move it inside the service and ensure that it will be reused from there but I would consider an extra refactor (that I would be willing to do in a separate PR if time allows it).
Moving it to the `evaluateCASSupplier` does not seem right. `evaluateCASSupplier` may not even be the first method to execute the login and it does not seem its responsibility.
For instance, wonder if we have a pending address to be verified and it just needs to call CAS and create a new site. This may not require a call to CAS from the `evaluateCASSupplier`.